### PR TITLE
Check for subreddit before building it's event data.

### PR DESCRIPTION
**Pull request by schwers**
Sometimes the subreddit fails to load, either due to the sub
being private, quarantined, or just a general API failure.
This fix prevents us from throwing an error while trying to build
screenview data when the subreddit data didn't load.

👓  @phil303 || @uzi 

This should be the source of the `cannot read property 'name' of undefined` errors in the tap.
e.g. log:
```js
{
  "column": "2661",
  "env": "CLIENT",
  "isAPIFailure": false,
  "line": "8",
  "message": "Error: Cannot read property 'name' of undefined",
  "reduxInfo": "Redux Action Stack: \"PLATFORM__SET_STATUS\", \"FAILED_PREFERENCES\", \"FAILED_SUBREDDIT\", \"RECEIVED_COMMENTS_PAGE\",{\"type\":\"FETCHING_SUBREDDIT\",\"name\":\"askscience\"}, \"VISITED_COMMENTS_PAGE\", \"PENDING_PREFERENCES\",{\"type\":\"FETCHING_COMMENTS_PAGE\",\"commentsPageId\":\"8604f797d0ada76e2901b80686ec36e64e455ea6\",\"commentsPageParams\":{id\":\"t3_54o513\"}}, {\"type\":\"PLATFORM__SET_PAGE\",\"payload\":{\"url\":\"/r/askscience/comments/54o513/is_it_possible_to_shrink_plastic_then_bring_it/\",\"urlParams\"{\"subredditName\":\"askscience\",\"postId\":\"54o513\",\"postTitle\":\"is_it_possible_to_shrink_plastic_then_bring_it\"},\"queryParams\":{},\"hashParams\":{},\"referrer\":\"\"}}, \"PLATFORM__NAVIGATE_TO_URL\", \"PLATFORM__SET_SHELL\"",
  "requestUrl": "/r/askscience/comments/54o513/is_it_possible_to_shrink_plastic_then_bring_it/",
  "stack": "TypeError: Cannot read property 'name' of undefined
at o(https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:8:2661)
    at u(https: //www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:20:15966)
    at a(https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:8:3264)
    at t.<anonymous>(https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:20:17301)
    at i (https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:31:23040)
    at s._invoke(https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:31:24478)
    ats.e.(anonymous function) [as next] (https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:31:23219)
    at r (https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:20:14901)
    at https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js:20:15008",
  "url": "https://www.redditstatic.com/mweb2x/ProductionClient.bb7975933943fe2cbf14.js",
  "userAgent": "Mozilla/5.0 (Linux; Android 6.0; LG-D852 Build/MRA58K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.124 Mobile Safari/537.36"
}
```

Minified source for top line of stack trace:
```js
    function o(e) {
        var t = e.platform.currentPage.urlParams.subredditName;
        if (!n.i(p.a)(t)) {
            var o = e.subreddits[t.toLowerCase()];
            return {
                sr_id: r(o.name), // name of undefined error happens here
                sr_name: o.uuid
            }
        }
        return {}
    }
```

### CODE REVIEW

**Comment by uzi** [comment]
:fish:

File: src/lib/eventUtils.js [hunk 1]
```diff
@@ -16,14 +16,15 @@ export function convertId(id) {
...
 
-  if (!isFakeSubreddit(subredditName)) {
-    const subreddit = state.subreddits[subredditName.toLowerCase()];
-    return {
-      sr_id: convertId(subreddit.name),
-      sr_name: subreddit.uuid,
-    };
-  }
-  return {};
+  if (!isFakeSubreddit(subredditName)) { return {}; }
```
Comment (phil303) [inline src/lib/eventUtils.js]: i dont think this logic is right. if it's not a fake subreddit, we _want_ to track it.

Comment (schwers) [inline src/lib/eventUtils.js]: derp, you're right, thanks phil

**Comment by phil303** [comment]
:nail_care:

File: src/lib/eventUtils.js (hunk 1 above)
Comment (uzi) [inline src/lib/eventUtils.js]: Booleans are hard.

**Comment by schwers** [comment]
💇

**Comment by phil303** [comment]
:fish: thanks!

**Pull request merged**